homeassistant/components/mqtt/mixins.py

```diff
@@ -470,3 +471,73 @@ async def device_info_discovery_update(self, config: dict):
     def device_info(self):
         """Return a device description for device registry."""
         return device_info_from_config(self._device_config)
+
+
+class MqttEntity(
+    MqttAttributes,
+    MqttAvailability,
+    MqttDiscoveryUpdate,
+    MqttEntityDeviceInfo,
+):
+    """Representation of an MQTT entity."""
+
+    def __init__(self, hass, config, config_entry, discovery_data):
+        """Init the MQTT Entity."""
+        self.hass = hass
+        self._unique_id = config.get(CONF_UNIQUE_ID)
+        self._sub_state = None
+
+        # Load config
+        self._setup_from_config(config)
+
+        # Initialize mixin classes
+        MqttAttributes.__init__(self, config)
```

Comment:
For the future: It would be good if we refactor the entity init methods and inheritance so we can just call `super().__init__` in each entity and mixin and have it work correctly.